api/indexer/client.go

```diff
@@ -89,24 +92,34 @@ func (c *Client) GetTx(ctx context.Context, txID ids.ID) (GetTxResponse, bool, e
 	// We use string parsing here because the JSON-RPC library we use may not
 	// allows us to perform errors.Is.
 	case err != nil && strings.Contains(err.Error(), ErrTxNotFound.Error()):
-		return GetTxResponse{}, false, nil
+		return GetTxResponse{}, nil, false, nil
 	case err != nil:
-		return GetTxResponse{}, false, err
+		return GetTxResponse{}, nil, false, err
 	}
 
-	return resp, true, nil
+	var tx *chain.Transaction
+	if parser != nil {
+		p := codec.NewReader(resp.TxBytes, consts.NetworkSizeLimit)
+		tx, err = chain.UnmarshalTx(p, parser.ActionCodec(), parser.AuthCodec())
+		if err != nil {
+			return GetTxResponse{}, nil, false, fmt.Errorf("failed to unmarshal tx %s: %w", txID, err)
+		}
+	}
+	return resp, tx, true, nil
 }
 
 func (c *Client) WaitForTransaction(ctx context.Context, txCheckInterval time.Duration, txID ids.ID) (bool, uint64, error) {
 	var success bool
 	var fee uint64
 	if err := jsonrpc.Wait(ctx, txCheckInterval, func(ctx context.Context) (bool, error) {
-		response, found, err := c.GetTx(ctx, txID)
+		response, _, found, err := c.GetTx(ctx, txID, nil)
 		if err != nil {
 			return false, err
 		}
-		success = response.Success
-		fee = response.Fee
+		if found && response.Result != nil {
+			success = response.Result.Success
+			fee = response.Result.Fee
+		}
```

Comment:
Do we need the check for result to be nil? We should guarantee this is never the case